### PR TITLE
Lexicon: record embeds with media

### PR DESCRIPTION
**Pull request by devinivy**
Replaces #708, to allow an images or external embed to appear alongside a record embed.

### CODE REVIEW

**Comment by pfrazee** [comment]
Eugh god, sorry -- all looks good except why `complexRecord` instead of `recordWithMedia`? Complex Record seems odd to me

File: packages/pds/src/app-view/services/feed/types.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import { View as ViewImage } from '../../../lexicon/types/app/bsky/embed/images'
+import { View as ViewImages } from '../../../lexicon/types/app/bsky/embed/images'
 import { View as ViewExternal } from '../../../lexicon/types/app/bsky/embed/external'
 import { View as ViewRecord } from '../../../lexicon/types/app/bsky/embed/record'
+import { View as ViewComplexRecord } from '../../../lexicon/types/app/bsky/embed/recordWithMedia'
```
Comment (dholms) [inline packages/pds/src/app-view/services/feed/types.ts:4]: not a big deal, but "complex record" hung around here

File: packages/pds/src/app-view/services/indexing/plugins/post.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { Record as PostRecord } from '../../../../lexicon/types/app/bsky/feed/po
 import { isMain as isEmbedImage } from '../../../../lexicon/types/app/bsky/embed/images'
 import { isMain as isEmbedExternal } from '../../../../lexicon/types/app/bsky/embed/external'
 import { isMain as isEmbedRecord } from '../../../../lexicon/types/app/bsky/embed/record'
+import { isMain as isEmbedComplexRecord } from '../../../../lexicon/types/app/bsky/embed/recordWithMedia'
```
Comment (dholms) [inline packages/pds/src/app-view/services/indexing/plugins/post.ts:8]: same here

File: packages/pds/src/app-view/services/indexing/plugins/post.ts [hunk 3]
```diff
@@ -76,34 +77,47 @@ const insertFn = async (
...
-      postUri: uri.toString(),
-      embedUri: record.uri,
-      embedCid: record.cid,
+  const embeds: (PostEmbedImage[] | PostEmbedExternal | PostEmbedRecord)[] = []
+  const postEmbeds = isEmbedComplexRecord(obj.embed)
+    ? [
+        { $type: lex.ids.AppBskyEmbedRecord, ...obj.embed.record },
+        obj.embed.media,
+      ]
+    : obj.embed
```
Comment (dholms) [inline packages/pds/src/app-view/services/indexing/plugins/post.ts:86]: can we make this an explicit check (`obj.embed !== undefined`)? just for readabilities sake? thought it was a value at first

File: packages/pds/src/repo/prepare.ts [hunk 4]
```diff
@@ -41,29 +45,35 @@ export const blobsForWrite = (record: any): PreparedBlobRef[] => {
...
-    if (embed?.$type === 'app.bsky.embed.images') {
-      const doc = lex.schemaDict.AppBskyEmbedImages
-      for (let i = 0; i < embed.images?.length || 0; i++) {
-        const img = embed.images[i]
+    const embeds = isComplexRecordEmbed(record.embed)
+      ? [
+          { $type: 'app.bsky.embed.record', ...record.embed.record },
+          record.embed.media,
+        ]
+      : record.embed
```
Comment (dholms) [inline packages/pds/src/repo/prepare.ts:55]: same here

Comment (devinivy) [reply packages/pds/src/repo/prepare.ts]: Extracted this into a function for readability in both places 👍

File: packages/pds/src/app-view/services/feed/types.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/app-view/services/feed/types.ts]: Ah, thanks for nabbing that.  Should be patched-up.

**Pull request merged**